src/main/java/com/github/emilshina/utils/CsvUtils.java

```diff
@@ -0,0 +1,40 @@
+package com.github.emilshina.utils;
+
+import com.github.emilshina.model.User;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+
+public class CsvUtils {
+
+    private CsvUtils() {
+        throw new UnsupportedOperationException("Illegal access to private constructor.");
+    }
+
+    public static <T> T csvToEntity(final String dataSource, final Class<T> entityClass) {
+
+        final String path = ClassLoader.getSystemResource(dataSource).getPath();
+
+        try (Stream<String> lines = Files.lines(Paths.get(path))) {
+            List<String> values = lines
+                    .filter(line -> line.startsWith("3099"))
+                    .flatMap(line -> (Arrays.stream(line.split(";"))))
+                    .collect(Collectors.toList());
+
+            return new User(Long.parseLong(values.get(0)), values.get(1), values.get(2), asList(values.get(3)));
```

Comment:
a good example of using reflection... you can't init a new object of generic type without reflection... and otherwise, your method won't be generic...